ios/IoReactNativeIntegrity.swift

```diff
@@ -1,11 +1,62 @@
 import DeviceCheck
 import CryptoKit
+import SwiftCBOR
 
 /// A class to interact with the DeviceCheck and CryptoKit frameworks to ensure the integrity of the device and app.
 @objc(IoReactNativeIntegrity)
 class IoReactNativeIntegrity: NSObject {
   private typealias ME = ModuleException
   
+  struct DecodedData: Codable {
+    var signature: [UInt8]
+    var authenticatorData: [UInt8]
+  }
+  
+  enum CBORDecodingError: Error {
+    case invalidFormat(String)
+  }
+  
+  /// Decodes a base64 encoded CBOR assertion into a JSON object.
+  /// - Parameters:
+  ///   - assertion: The CBOR assertion to decode.
+  /// - Returns: A JSON object representing the decoded assertion.
+  @objc(decodeAssertion:withResolver:withRejecter:)
+  func decodeAssertion(
+    assertion: String,
+    resolve: @escaping RCTPromiseResolveBlock,
+    reject: @escaping RCTPromiseRejectBlock
+  ) -> Void {
+    // Attempt to decode the CBOR data
+    do {
+      // Convert base64 encoded assertion string to Data
+      guard let data = Data(base64Encoded: assertion) else {
+        throw CBORDecodingError.invalidFormat("Expected base64 endoded string")
+      }
+      
+      let cborData = [UInt8](data)
+      let decodedValue = try CBOR.decode(cborData)
+      
+      // Ensure the decoded value is a map
+      guard case let .map(map) = decodedValue else {
+        throw CBORDecodingError.invalidFormat("Expected CBOR map")
+      }
+      
+      // Extract the signature and authenticatorData from the map
+      guard case let .byteString(signatureBytes)? = map["signature"],
+            case let .byteString(authenticatorDataBytes)? = map["authenticatorData"] else {
+        throw CBORDecodingError.invalidFormat("Expected signature and authenticatorData in the CBOR map")
```

Comment:
Given that we are writing this library directly into the bridge code, should we remove this `CBORDecodingError` and just reject the promise with our `ME` object and the specified message? We could then get rid of the custom error.

Reply:
I see it as a complication we don't need but it's fine, we can leave it like this.